lib/streamlit/elements/write.py

```diff
@@ -437,7 +437,16 @@ def flush_buffer():
                 dot = vis_utils.model_to_dot(arg)
                 self.dg.graphviz_chart(dot.to_string())
             elif isinstance(
-                arg, (dict, list, SessionStateProxy, UserInfoProxy, QueryParamsProxy)
+                arg,
+                (
+                    dict,
+                    list,
+                    SessionStateProxy,
+                    UserInfoProxy,
+                    QueryParamsProxy,
+                    StreamlitHeaders,
+                    StreamlitCookies,
```

Comment:
Actually, I thought about it during implementation and eventually came to the conclusion that it is better not to override `st.write(st.context)` output via json (because context is an object with properties, not AttrDict / Dict-like object). 

Our default handling of class objects works great for this case, giving all the information that is needed. (The actual docstring would be changed, but the appearance would look like this). 

<img width="827" alt="Screenshot 2024-07-05 at 13 50 55" src="https://github.com/streamlit/streamlit/assets/6664805/1a99de1e-c22c-4eed-8028-955af444ccbf">